docs/core/logging.md

```diff
@@ -178,148 +224,286 @@ Powertools for AWS Lambda (Java) extends the functionality of Log4J. Below is an
+| **error**         | Map    | `{ "name": "InvalidAmountException", "message": "Amount must be superior to 0", "stack": "at..." }` | Eventual exception (e.g. when doing `logger.error("Error", new InvalidAmountException("Amount must be superior to 0"));`)  |
+
+### Log messages as JSON
+By default, `message` is logged as a `String` (e.g `"message": "The message"`). When logging JSON content, 
+you may want to avoid the escaped String (`"message:"{\"key\":\"value\"}"`) for better readability. 
+You can use `LoggingUtils.logMessagesAsJson(true)` to enable this programmatically. 
+
+=== "PaymentFunction.java"
+
+    ```java hl_lines="14 15 17-20"
+    import static software.amazon.lambda.powertools.utilities.EventDeserializer.extractDataFrom;
+    import software.amazon.lambda.powertools.logging.LoggingUtils;
+    import software.amazon.lambda.powertools.utilities.JsonConfig;
+    // ... other imports
+
+    public class PaymentFunction implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
     
+        private static final Logger LOGGER = LoggerFactory.getLogger(PaymentFunction.class);
+        
         @Logging
         public APIGatewayProxyResponseEvent handleRequest(final APIGatewayProxyRequestEvent input, final Context context) {
-         ...
+            Order order = extractDataFrom(input).as(Order.class);
+
+            // logged as a String
+            LOGGER.debug("{}", JsonConfig.get().getObjectMapper().writeValueAsString(order));
+
+            // Logged as JSON
+            LoggingUtils.logMessagesAsJson(true);
+            LOGGER.debug("{}", JsonConfig.get().getObjectMapper().writeValueAsString(order));
+            LoggingUtils.logMessagesAsJson(false);
+
+            // ...
         }
     }
     ```
 
+=== "Order.java"
+
+    ```java 
+    public class Order {
+        private String id;
+        private Date date;
+        private Double amount;
+    }
+    ```
+
+=== "Example CloudWatch Logs"
+
+    ```json hl_lines="3 9-13"
+	{
+		"level": "DEBUG",
+	  	"message": "{\"id\":\"435iuh2j3hb4\", \"date\":\"2023-12-01T14:48:59\", \"amount\":435.5}",
+		"timestamp": "2023-12-01T14:49:19.293Z",
+	  	"service": "payment",
+	}
+	{
+		"level": "DEBUG",
+	  	"message": {
+            "id": "435iuh2j3hb4", 
+            "date": "2023-12-01T14:48:59", 
+            "amount":435.5
+        },
+		"timestamp": "2023-12-01T14:49:19.312Z",
+	  	"service": "payment",
+	}
+    ```
+
+You can also achieve this more broadly for all JSON messages (see advanced configuration for [log4j](#log-messages-as-json-1) & [logback](#log-messages-as-json-2)).
+
+## Additional structured keys
+
+### Logging Lambda context information
+The following keys will also be added to your structured logs (unless [configured otherwise](#more-customization-1)):
```

Comment:
I think this link goes nowhere?
Also might be worth making it _super explicit_ that these keys are added to all logs emitted, until they are removed again. You mention it later on in "Logging additional keys" I think but it applies to everything in this section

Reply:
the link works...
```suggestion
The following keys will also be added to all your structured logs (unless [configured otherwise](#more-customization-1)):
```